xchange-bx/src/main/java/org/knowm/xchange/bx/BxAdapters.java

```diff
@@ -0,0 +1,164 @@
+        if (orderType == OrderType.BID) {
+            return BUY;
+        } else if (orderType == OrderType.ASK) {
+            return SELL;
+        } else {
+            throw new ExchangeException("Unsupported order type");
+        }
+    }
+
+    private static OrderType adaptBxOrderType(String orderType) {
+        OrderType result = null;
+        if (orderType.equals(BUY)) {
+            result = OrderType.BID;
+        } else if (orderType.equals(SELL)) {
+            result = OrderType.ASK;
+        }
+        return result;
+    }
+
+    public static OpenOrders adaptOpenOrders(BxOrder[] orders) {
+        List<LimitOrder> limitOrders = new ArrayList<>();
+        for (BxOrder order : orders) {
+            limitOrders.add((LimitOrder) adaptOrder(order));
+        }
+        return new OpenOrders(limitOrders);
+    }
+
+    private static Date adaptDate(String date) {
+        DateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Date result;
+        try {
+            result = format.parse(date);
+        } catch (ParseException e) {
+            throw new ExchangeException("Date/time parse error: " + e.getMessage());
+        }
+        return result;
+    }
+
+    public static Order adaptOrder(BxOrder bxOrder) {
+        return new LimitOrder(
+                adaptBxOrderType(bxOrder.getOrderType()),
+                bxOrder.getAmount(),
+                BxUtils.translateBxCurrencyPair(bxOrder.getPairingId()),
+                bxOrder.getOrderId(),
+                adaptDate(bxOrder.getDate()),
+                bxOrder.getRate()
+        );
+    }
+
+    public static Wallet adaptWallet(Map<String, BxBalance> currencies) {
+        List<Balance> balances = new ArrayList<>();
+        for (String record : currencies.keySet()) {
+            Balance balance = new Balance(
+                    adaptCurrency(record),
+                    currencies.get(record).getTotal(),
+                    currencies.get(record).getAvailable(),
+                    currencies.get(record).getOrders(),
+                    BigDecimal.ZERO,
+                    BigDecimal.ZERO,
+                    currencies.get(record).getWithdrawals(),
+                    currencies.get(record).getDeposits()
+            );
+            balances.add(balance);
+        }
+        return new Wallet(balances);
+    }
+
+    private static Currency adaptCurrency(String currency) {
+        return BxUtils.translateCurrency(currency);
+    }
+
+    public static UserTrades adaptUserTrades(BxTradeHistory[] histories) {
+        List<UserTrade> trades = new ArrayList<>();
+        for (BxTradeHistory history : histories) {
+            trades.add(adaptUserTrade(history));
+        }
+        return new UserTrades(trades, Trades.TradeSortType.SortByTimestamp);
+    }
+
+    private static UserTrade adaptUserTrade(BxTradeHistory history) {
+        return new UserTrade(
+                OrderType.BID,
```

Comment:
You use BID for all trades. You have to use BID for history.getAmount() > 0, and ASK for history.getAmount() < 0.
Also exchange returns pair of transactions that have to be combined into `UserTrade`